src/battle_terastal.c

```diff
@@ -63,16 +63,19 @@ bool32 CanTerastallize(u32 battler)
 {
     u32 holdEffect = GetBattlerHoldEffect(battler, FALSE);
 
-    // Check if Player has Tera Orb and has charge.
-    if (!TESTING && !CheckBagHasItem(ITEM_TERA_ORB, 1))
-        return FALSE;
-
-    if (!TESTING 
-     && !(B_FLAG_TERA_ORB_NO_COST != 0 && FlagGet(B_FLAG_TERA_ORB_NO_COST))
-     && (battler == B_POSITION_PLAYER_LEFT || (!(gBattleTypeFlags & BATTLE_TYPE_MULTI) && battler == B_POSITION_PLAYER_RIGHT)))
+    if (GetBattlerSide(battler) == B_SIDE_PLAYER)
     {
-        if (B_FLAG_TERA_ORB_CHARGED != 0 && !FlagGet(B_FLAG_TERA_ORB_CHARGED))
+        // Check if Player has Tera Orb and has charge.
+        if (!TESTING && !CheckBagHasItem(ITEM_TERA_ORB, 1))
             return FALSE;
+
+        if (!TESTING
```

Comment:
Since we check `!TESTING` twice, shouldn't we just rewrite it so it only needs to be checked once?

Reply:
Also, this touches the same code as #4863. Does this supersede that PR, too?

Reply:
No, it doesn't. Only thing I did was wrap it around the player check but I agree with you that it should be somehow simplified. In general I think that the whole sequence of checks is very weird which I tried to simplify in #4863. In general though I would prefer it if this pr gets merged (ideally first) and the rest is handled in the old pr.